account_move_line_product_category/models/account_move_line.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2018 Jacques-Etienne Baudoux (BCIM sprl) <je@bcim.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMoveLine(models.Model):
+
+    _inherit = "account.move.line"
+
+    categ_id = fields.Many2one(
+        comodel_name="product.category",
+        string="Product Category",
+        index="btree_not_null",
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get("product_id") and "categ_id" not in vals:
+                product = self.env["product.product"].browse(vals["product_id"])
+                vals["categ_id"] = product.categ_id.id
+        return super().create(vals_list)
+
+    def write(self, vals):
+        if "product_id" in vals and "categ_id" not in vals:
+            if vals.get("product_id"):
+                product = self.env["product.product"].browse(vals["product_id"])
+                vals["categ_id"] = product.categ_id.id
+            else:
+                vals["categ_id"] = False
+        return super().write(vals)
```

Comment:
Maybe I am missing something, but couldn't this be simplified by a stored related?
```suggestion
    categ_id = fields.Many2one(
        related="product.category_id"
        store=True,
        index="btree_not_null",
    )
```
I agree that in that in that case we cannot change the value as your code is permitting, but if it is really what you want, maybe a compute store=True readonly=False could also make the job.
If you agree with it, we should also maybe consider running a SQL after installation in a `pre_init` hook instead of letting the ORM take care of this the first time the field is used.

Reply:
Hey @AaronHForgeFlow sorry I did not see your proposals before reviewing. As you can read I agree with your suggestion 🙂

Reply:
you don't want the account move lines to change in case the category on the product is modified

Reply:
It could maybe be simplified with a compute readonly=False that depends on product_id but not product_id.category_id?